code/game/gamemodes/events/blob.dm

```diff
@@ -0,0 +1,512 @@
+/*
+	The blob is a horrible acidic slime creature that eats through airlocks and expands infinitely.
+	The rate of expansion slows down as it grows, so it is ultimately soft-capped
+
+	Its attacks deal light burn damage, but spam many hits. They deal a lot of damage by splashing acid
+	onto victims, allowing acidproof gear to provide some good protection
+
+	Blobs are very vulnerable to fire and lasers. Flamethrower is the recommended weapon, and
+	In an emergency, a plasma canister and a lighter will bring a quick end to a blob
+*/
+
+/datum/storyevent/blob
+	id = "blob"
+	name = "Blob"
+
+
+	event_type = /datum/event/blob
+	event_pools = list(EVENT_LEVEL_MAJOR = POOL_THRESHOLD_MAJOR*1.2)
+	tags = list(TAG_COMBAT, TAG_DESTRUCTIVE, TAG_NEGATIVE)
+//============================================
+
+/datum/event/blob
+	announceWhen	= 12
+
+	var/obj/effect/blob/core/Blob
+
+/datum/event/blob/announce()
+	level_seven_announcement()
+
+/datum/event/blob/start()
+	var/area/A = random_ship_area(TRUE)
+	var/turf/T = A.random_space()
+	if(!T)
```

Comment:
Would it announce the blob alert if blob didn't spawn?

Reply:
nope thats what the kill proc is for. it aborts the event